components/formats-bsd/src/loci/formats/in/OMETiffReader.java

```diff
@@ -1056,7 +1063,14 @@ public MetadataStore getMetadataStoreForConversion() {
     int realSeries = getSeries();
     for (int i=0; i<getSeriesCount(); i++) {
       setSeries(i);
```

Comment:
It looks the code generation improvements forces us to revisit the API specifically added in https://github.com/openmicroscopy/bioformats/commit/de5dbaf9baea1b84ee34909dcc3cdcc518fe2613.

A couple of points that might require clarification in particular:
- is `BinData.BigEndian` still a requirement for converting file formats or can we use the MetadataStore populated by the OMETiffReader directly with an `IFormatWriter`
- similarly is there still a need for a converted OME-XML when displaying in `ImageInfo` especially now the validation happens on read and no longer after conversion with https://github.com/openmicroscopy/bioformats/pull/2456 ?

As this is public API, I would not drop `getMetadataStoreForDisplay()/getMetadataStoreForConversion`() but deprecating them would be sensible if there is no more clear usage for them.